README.md

```diff
@@ -90,85 +90,197 @@ Here the list of existing promise adapters:
 This bundle can be use with [GraphQLBundle](https://github.com/overblog/GraphQLBundle).
 Here an example:
 
-* Bundle config
+* First create your service. We will use the Webonyx promise adapter
 
 ```yaml
-#graphql
+#config/services.yaml
+services:
+    graphql_promise_adapter:
+        class: Overblog\DataLoader\Promise\Adapter\Webonyx\GraphQL\SyncPromiseAdapter
+        public: true
+        
+    
+    ###
+    # Force the overblog promise adapter to use the webonyx adapter
+    ###
+    Overblog\PromiseAdapter\PromiseAdapterInterface:
+        class: Overblog\PromiseAdapter\Adapter\WebonyxGraphQLSyncPromiseAdapter
+        arguments:
+            - "@graphql_promise_adapter"
+    
+    ###
+    # Add magic configuration to load all your dataLoader
+    ###
+    App\Loader\:
+        resource: "../src/Loader/*"
+```
+
+* Now configure the packages
+
+```yaml
+#config/packages/graphql.yaml
 overblog_graphql:
-    definitions:
-        schema:
-            query: Query
+    #[...]
     services:
-        promise_adapter: "webonyx_graphql.sync_promise_adapter"
+        promise_adapter: "graphql_promise_adapter"
 
-#dataloader
+#config/packages/dataloader.yaml
 overblog_dataloader:
     defaults:
-        promise_adapter: "overblog_dataloader.webonyx_graphql_sync_promise_adapter"
-    loaders:
-        ships:
-            alias: "ships_loader"
-            batch_load_fn: "@app.graph.ships_loader:all"
+        promise_adapter: "graphql_promise_adapter"
 ```
 
-* Batch loader function
+* Create an abstract class GenericDataLoader
 
-```yaml
-services:
-    app.graph.ship_repository:
-        class: AppBundle\Entity\Repository\ShipRepository
-        factory: ["@doctrine.orm.entity_manager", getRepository]
-        arguments:
-            - AppBundle\Entity\Ship
+```php
+<?php
 
-    app.graph.ships_loader:
-        class: AppBundle\GraphQL\Loader\ShipLoader
-        arguments:
-            - "@overblog_graphql.promise_adapter"
-            - "@app.graph.ship_repository"
+namespace App\Loader;
+
+use Overblog\DataLoader\DataLoader;
+use Overblog\PromiseAdapter\PromiseAdapterInterface;
+
+abstract class AbstractDataLoader extends DataLoader
+{
+    /**
+     * @param PromiseAdapterInterface $promiseAdapter
+     */
+    public function __construct(PromiseAdapterInterface $promiseAdapter) {
+        parent::__construct(
+            function ($ids) use ($promiseAdapter) {
+                return $promiseAdapter->createAll($this->find($ids));
+            },
+            $promiseAdapter
```

Comment:
So you are creating a callable?  The batch_load_fn option is supposed to take care of this.  Did this ever work? @mcg-web we're a bit lost I think.

Reply:
see my example here #22

Reply:
So I went around the barn the long way, and concluded that while the original example isn't terrific, this new one is doing it wrong.  

You are not supposed to inherit from Dataloader -- this bundle will create services which are instances of Dataloader, and those services will call the callable you define in "batch_load_fn".